art/attacks/svm_attack.py

```diff
@@ -0,0 +1,264 @@
+
+        if not isinstance(classifier, ScikitlearnSVC):
+            raise TypeError('Classifier must be a SVC')
+        if isinstance(self.classifier._model, LinearSVC):
+            self.classifier = ScikitlearnSVC(model=SVC(C=self.classifier._model.C, kernel='linear'),
+                                             clip_values=self.classifier.clip_values)
+            self.classifier.fit(x_train, y_train)
+        elif not isinstance(self.classifier._model, SVC):
+            raise NotImplementedError("Model type '{}' not yet supported".format(type(self.classifier._model)))
+
+        self.step = step
+        self.eps = eps
+        self.x_train = x_train
+        self.y_train = y_train
+        self.x_val = x_val
+        self.y_val = y_val
+        self.max_iters = max_iters
+        self.set_params(**kwargs)
+
+    def generate(self, x, **kwargs):
+        """
+        Iteratively finds optimal attack points starting at values at x
+
+        :param x: An array with the points that initialize attack points.
+        :type x: `np.ndarray`
+        :return: An array holding the adversarial examples.
+        :rtype: `np.ndarray`
+        """
+
+        if 'y' in kwargs:
+            y_attack = kwargs['y']
+        else:
+            y_attack = self.classifier.predict(x)
+
+        num_poison = len(x)
+        if num_poison == 0:
+            raise ValueError("Must input at least one poison point")
+        num_features = len(x[0])
+        train_data = np.copy(self.x_train)
+        train_labels = np.copy(self.y_train)
+        classifier = self.classifier
+        all_poison = []
+        for attack_point, attack_label in zip(x, y_attack):
+            poison, classifier = generate_attack_point(classifier, train_data, train_labels, self.x_val, self.y_val,
+                                                       attack_point, attack_label, self.step, self.eps, self.max_iters)
+            all_poison.append(poison)
+            train_data = np.vstack([train_data, poison])
+            train_labels = np.vstack([train_labels, attack_label])
+
+        self.classifier = classifier
+        return np.array(all_poison).reshape((num_poison, num_features))
+
+    def set_params(self, **kwargs):
+        """
+        Take in a dictionary of parameters and apply attack-specific checks before saving them as attributes.
+
+        :param kwargs: a dictionary of attack-specific parameters
+        :type kwargs: `dict`
+        :return: `True` when parsing was successful
+        """
+        super(SVMAttack, self).set_params(**kwargs)
+        if self.step <= 0:
+            raise ValueError("Step size must be strictly positive")
+        if self.eps <= 0:
+            raise ValueError("Value of eps must be strictly positive")
+        if self.max_iters <= 1:
+            raise ValueError("Value of max_iters must be strictly positive")
+
+
+def generate_attack_point(classifier, x_train, y_train, x_val, y_val, x_attack, y_attack, step, eps, max_iters):
```

Comment:
Could you make this an instance method belonging to the class?
If yes, remove the redundant arguments and use `self`.
It looks like this function is called only from within this class. If you don't intend users to call this function, you could add a underscore `_` at the beginning of the method name.

Reply:
I thought it could be used as a convenience function and is called explicitly from the notebook.

Reply:
I see. To follow the pattern of the other attacks in ART, could we make it an instance method and call it in the notebook as something like ```
attack = SVMAttack()
attack_point, poisoned = attack.generate_attack_point()
```